test/TensorFlowRuntime/tensor_group.swift

```diff
@@ -0,0 +1,78 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-build-swift -swift-version 5 -g %s -o %t/a.out
+// RUN: %target-codesign %t/a.out
+// RUN: %target-run %t/a.out
+// REQUIRES: executable_test
+//
+// `TensorGroup` tests.
+
+import TensorFlow
+import StdlibUnittest
+
+var TensorGroupTests = TestSuite("TensorGroup")
+
+extension TensorDataType : Equatable {
+  public static func == (lhs: TensorDataType, rhs: TensorDataType) -> Bool {
+    return Int(lhs._cDataType.rawValue) == Int(rhs._cDataType.rawValue)
+  }
+}
+
+struct Empty : TensorGroup {}
```

Comment:
It would be great if `TensorArrayProtocol` conformances are thoroughly tested too.

Reply:
Yes, `TensorArrayProtocol` tests please!
I think this PR will be ready for merging after tests for all derived implementations are added.

Reply:
Haha I actually implemented some and discovered I'm missing out the derivation for `_tensorHandleCount` because it was implemented for `TensorGroup` by default. I'll add that later today. :)